src/mlpack/tests/main_tests/dbscan_test.cpp

```diff
@@ -0,0 +1,476 @@
+{
+  arma::mat inputData;
+  if (!data::Load("iris.csv", inputData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  SetInputParam("input", inputData);
+  SetInputParam("epsilon", (double) 1.0);
+
+  mlpackMain();
+
+  arma::Row<size_t> output1;
+  output1 = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
+  bindings::tests::CleanMemory();
+
+  CLI::GetSingleton().Parameters()["input"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["epsilon"].wasPassed = false;
+
+  SetInputParam("input", inputData);
+  SetInputParam("epsilon", (double) 0.5);
+
+  mlpackMain();
+
+  arma::Row<size_t> output2;
+  output2 = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
+  BOOST_REQUIRE_GT(arma::accu(output1 != output2), 1);
+}
+
+/**
+ * Check that the cluster assignment is different for different
+ * values of Min Size
+ */
+BOOST_AUTO_TEST_CASE(DBSCANDiffMinSizeTest)
+{
+  arma::mat inputData;
+  if (!data::Load("iris.csv", inputData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  SetInputParam("input", inputData);
+  SetInputParam("epsilon", (double) 0.4);
+  SetInputParam("min_size", (int) 5);
+
+  mlpackMain();
+
+  arma::Row<size_t> output1;
+  output1 = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
+  bindings::tests::CleanMemory();
+
+  CLI::GetSingleton().Parameters()["input"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["epsilon"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["min_size"].wasPassed = false;
+
+  SetInputParam("input", inputData);
+  SetInputParam("epsilon", (double) 0.5);
+  SetInputParam("min_size", (int) 40);
+
+  mlpackMain();
+
+  arma::Row<size_t> output2;
+  output2 = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
+  BOOST_REQUIRE_GT(arma::accu(output1 != output2), 1);
+}
+
+/**
+ * Check that number of noise points increases as
+ * the value of epsilon decreases.
+ */
+BOOST_AUTO_TEST_CASE(DBSCANEpsilonNoisePointsTest)
+{
+  arma::mat inputData;
+  if (!data::Load("iris.csv", inputData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  SetInputParam("input", inputData);
+  SetInputParam("epsilon", (double) 1.0);
+
+  mlpackMain();
+
+  arma::Row<size_t> output1;
+  output1 = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
+  int noisePointsOutput1 = 0;
+
+  for (size_t i = 0; i < output1.n_elem; ++i)
+  {
+    if (output1[i] == 18446744073709551615)
+    {
+      noisePointsOutput1++;
+    }
+  }
+
+  bindings::tests::CleanMemory();
+
+  CLI::GetSingleton().Parameters()["input"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["epsilon"].wasPassed = false;
+
+  SetInputParam("input", inputData);
+  SetInputParam("epsilon", (double) 0.4);
+
+  mlpackMain();
+
+  arma::Row<size_t> output2;
+  output2 = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
+  int noisePointsOutput2 = 0;
+
+  for (size_t i = 0; i < output2.n_elem; ++i)
+  {
+    if (output2[i] == 18446744073709551615)
+    {
```

Comment:
It think it would be better use `arma::accu(A==B)` here. And moreover instead of mentioning this big dirty number here may be you can use some kind of numeric limit as noise may not always have this value . Please check this at other places too.

Reply:
I got the same number for noise points on all datasets I ran DBSCAN on. I'm not sure if noise can have a different value..

Reply:
Ya, you are right as garbage value depends on the datatype. So, it will remain same for a particular datatype.

Reply:
Since, you are using iris dataset so the assignments can't be greater than 3. So, may be you case use something like `noisePointOutput2 = arma::accu(output2 < 0 && output2 > 2)` to calculate `noisePoints` and add a small comment about the same. Let me know what you think.

Reply:
I think there is a misunderstanding. This is unsupervised learning. Assignments can be as many as possible with suitable input options. DBSCAN just forms clusters. If you give min size of clusters to be 0 and radius of search(Epsilon) to be very less as well, you can get as many clusters as the number of initial points.
Something like `output2 > inputData.n_cols` would be more appropriate.

Reply:
You are right, that's what I am trying to specify that there will always be a upper limit, as in case of assignments it will be 3 and as you said in clusters it will be inputData.n_cols. Let me know if it clears anything.

Reply:
Assignments are the same as the number of clusters in DBSCAN.
I'll take care of it! I also didn't want to use the garbage value. Thanks for your help!

Reply:
Ya, you are right, I just wanted to reflect the idea of using arma::accu().

Reply:
It's absolutely fine! I was able to think of another test and remove those sad garbage values! Thank you for your help!